src/cpp/src/continuous_batching_pipeline.cpp

```diff
@@ -87,7 +87,10 @@ class ContinuousBatchingPipeline::Impl {
             const auto& request = *requests_iterator;
             if(request->has_finished() || request->out_of_memory() || request->handle_dropped()) {
                 for (const auto& sequence: request->get_sequences()) {
-                    m_scheduler->free_sequence(sequence->get_id());
+                    size_t seq_id = sequence->get_id();
+                    if (m_scheduler->has_block_table(seq_id)) {
+                        m_scheduler->free_sequence(seq_id);
```

Comment:
in what cases we have such situation? is not it a bug in calling code?

Reply:
Basically without these checks I encounter failures inside the `free_sequence` (since I rewrote it to throw in case of double free). It sure looks like bugs in the calling code, but I didn't know the invariants expected at caller sites to correctly fix them. One occurs when a `FINISHED` sequence is freed at this place, and the other happens in `scheduler.hpp` code (I added a similar check there). I could fix it if you could tell me the invariants related to `FINISHED`/non-`FINISHED` sequences w.r.t. block table contents.